### ISSUE TITLE
ActiveMotifSlider widget limitsFromDb attribute not working with .BDL

### ISSUE THREAD

**Issue by slominskir** [JeffersonLab/wedm#20]
The limitsFromDb attribute in an EDL file indicates that an ActiveMotifSlider widget should obtain min and max scale bounds from EPICS.  This is an alternative to explicit scaleMin and scaleMax attributes.  Support for limitsFromDb is provided, but does not work if the channel field contains .BDL.  This issue may result in the "knob" on the slider appearing in unexpected locations on the display, outside of the widget itself.

**Comment by slominskir** [comment]
Example screen: 

https://epicsweb.jlab.org/wedm/screen?edl=/cs/jtabs/edm/magnet/macro/trim_expert.edl&%24(MAG)=MQS1L10

The upper and lower display limit fields are what we need as shown here:

https://www.javadoc.io/doc/org.epics/jca/latest/gov/aps/jca/dbr/DBR_GR_Double.html

epics2web doesn't monitor TIME, CTRL, or GR records though, so the specific needed fields must be explicitly monitored (.LOPR and .HOPR ?)

**Comment by slominskir** [comment]
Looks like we already have code in place to handle this:

https://github.com/JeffersonLab/wedm/blob/master/src/main/webapp/resources/js/screen.js#L585

But it is not be working as expected.   Gary mentioned something about limits may not always come from .LOPR/.HOPR and sometimes come from .HOPB and LOPB fields instead with .BDL values.   Maybe this is the real issue?

**Comment by slominskir** [comment]
Looks like the problem is with how WEDM handles channel name suffixes.   Specifically, with the above example display WEDM is searching for the base channel name from the PV name by looking for the period (.) and removing it all that follows.  Then .HOPR and .LOPR are appended.  This results in the equivalent of the command line:

```
caget MQS1L10.HOPR
MQS1L10.HOPR                  6
```

However, it should be:

```
caget MQS1L10.BDL.HOPR
MQS1L10.BDL.HOPR              24.1993
```

Notice nested suffixes in the name.

**Comment by slominskir** [comment]
Looks like .BDL is a JLab-custom field that relies on two additional custom fields for graphical bounds: .HOPB and .LOPB.    This works in native EDM because a custom GR record is created on the IOC to hide this and EDM monitors GR records (or subclasses).   epics2web only monitors simple records.

**Comment by slominskir** [comment]
Partially addressed in release [1.3.2](https://github.com/JeffersonLab/wedm/releases/tag/v1.3.2).   The slider knob will no longer leave widget bounds, but the bounds will still be numerically incorrect for JLab .BDL fielsd and knob position will likely be incorrect (at one of the extremes).  This seems "good enough" since WEDM is read-only and slider knobs are really just visual candy - virtually always accompanied by text box indicating numeric value.  The cost to add custom logic to WEDM for this custom field is high due to the way WEDM is implemented  .HOPR and .LOPR are treated the same globally for all widgets - this custom code would complicate / interfere with all widgets (the real cost is the added complexity, not adding code).  Feel free to press the issue if needed, but at this time, the partial fix seems sufficient.
